### ISSUE TITLE
bug(mat-expansion-panel): header position is incorrect when call elementInsideMatExpansionPanel.scrollIntoView(true)

### ISSUE THREAD

**Issue by stargazer8tw** [angular/components#21159]
#### Reproduction

- Components: https://angular-mat-expansion-panel-header-gone-after-scrollintoview.stackblitz.io

Steps to reproduce:

1. open [StackBlitz reproduce example](https://angular-mat-expansion-panel-header-gone-after-scrollintoview.stackblitz.io)
2. scroll to bottom
3. click `Test ScrollIntoView` button.

or.

1. Add `<span class="target">test</span>` inside `mat-expansion-panel` tag.
2. Set `mat-expansion-panel` `expanded="false"`.
3. Call `elementInsideMatExpansionPanel.scrollIntoView(true)`, e.g. `document.querySelector('.bug-target').scrollIntoView(true)`.
 
#### Expected Behavior

Viewport scrolls to `mat-expansion-panel` with `mat-expansion-panel-header` display, and element inside `mat-expansion-panel` remains hidden.

#### Actual Behavior

Viewport scrolls to `mat-expansion-panel` and causes `mat-expansion-panel-header` x-position changed and `mat-expansion-panel-header` is not display.

#### Environment

- Angular: 11.0.1
- CDK/Material: 11.0.1
- Browser(s): Chrome Version 87.0.4280.66 (Official Build) (64-bit)
- Operating System (e.g. Windows, macOS, Ubuntu): Windows, Ubutntu

#### Workaround

Call `window.scroll` or `scrollTo` function, see [StackBlitz reproduce example](https://angular-mat-expansion-panel-header-gone-after-scrollintoview.stackblitz.io)

**Comment by crisbeto** [comment]
Double-checking that I understood the issue correctly: the problem you're reporting is that the header text disappears when pressing the `Test ScrollIntoView`, correct?

**Comment by stargazer8tw** [comment]
Yes, that's correct. The header text disappears when pressing the `Test ScrollIntoView`.

**Comment by crisbeto** [comment]
Better repo from #21622: https://stackblitz.com/edit/angular-yg9mb5.

**Comment by jazpearson** [comment]
We've just run into this ourselves....

**Comment by skantemneni** [comment]
I ran into this same issue on latest version of Angular Material...."@angular/material": "^16.1.3".  Hope there is an answer.  I defaulted to scrolling to just the Expansion Panel as a while.  Not any particular element within.

**Comment by WaveBeat** [comment]
Would be nice to get a fix for this

**Comment by cvoge** [comment]
For me, this was happening when closing a MatDialog window.

I created a Directive as a work-around:

`import { Directive, ElementRef, Renderer2 } from '@angular/core';

@Directive
({selector: '[panelHeaderDirectv]',})

export class PanelHeaderDirective {
  constructor(private el: ElementRef, private renderer: Renderer2) { }

  adjustPanelHeader() {
    const element = this.el.nativeElement;
    element.scrollIntoView({block: 'nearest', inline: 'nearest'});
  }
}`

-----------
html
`<mat-expansion-panel class="expansionPanel" id="detailPanel" *ngIf="showDetailsTab" [expanded]="fn()" (afterExpand)="fn($event)">
    <mat-expansion-panel-header class="panelHeader" panelHeaderDirectv>
        <mat-panel-title class="matTitle">
            {{detailTitle}}
        </mat-panel-title>
    </mat-expansion-panel-header>
    <div style="font-size: 12.5px">
...
...
...`

Typescript
- Define directive in module.ts
- Use @ViewChild to see directive
`dialogRef.afterClosed().subscribe(result => {
          code();
          code();
          this.panelHeaderDirectv.adjustPanelHeader();
        }
      });`
